### PR TITLE
feat!: expose KDBX4 saving behind feature flag

### PR DESCRIPTION
**Pull request by sseemayer**
- [x] Create feature flag and change visibility
- [x] Update docs with disclaimers
- [x] Fix issues with opening created files
- [x] Test drive the feature somewhat to prevent catastrophic data losses

### CODE REVIEW

**Comment by sseemayer** [comment]
Right now, all tests pass but the generated database file from the readme example cannot be opened by KeePassXC, so more testing is definitely needed.

**Comment by sseemayer** [comment]
Made some progress by stepping through the KeePassXC sourcecode - at the end of `VariantDictionary` dumping, there needs to be a null byte. I am now running into errors parsing the inner header.

File: src/variant_dictionary.rs [hunk 1]
```diff
@@ -141,6 +141,9 @@ impl VariantDictionary {
             data.extend_from_slice(&field_buffer);
         }
 
+        // signify end of variant dictionary
+        data.push(0);
```
Comment (louib) [inline src/variant_dictionary.rs:145]: @sseemayer does that mean that we were also not verifying that when parsing? Should we add it?

Comment (sseemayer) [reply src/variant_dictionary.rs:145]: Yes, we should. There will be probably more details like this that will be uncovered before files generated by us can be opened by other KeePass implementations.

Comment (sseemayer) [reply src/variant_dictionary.rs:155]: resolved with da4c9d5 .

**Comment by sseemayer** [comment]
I can now use the project to parse my personal KeePass database, dump it again and re-open it in KeePassXC. From diff-ing the inner XML file, I know that some parts are still missing such as DeletedObjects, but putting it behind an experimental feature flag makes sense so we can gather some additional user feedback.

**Pull request merged**